src/apps/user_auth/models.py

```diff
@@ -0,0 +1,66 @@
+from django.utils import timezone
+
+from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
+from django.contrib.auth.models import PermissionsMixin
+from django.db import models
+
+
+class UserManager(BaseUserManager):
+    def _create_user(self, uid, password, **extra_fields):
+        """
+        Creates and saves a User with the given id and password,
+        :param id:
+        :param password:
+        :param extra_fields:
+        :return:
+        """
+        if not uid:
+            raise ValueError('The given uid must be set')
+        user = self.model(uid=uid, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(self, uid, password=None, **extra_fields):
+        extra_fields.setdefault('is_staff', False)
+        extra_fields.setdefault('is_superuser', False)
+        return self._create_user(uid, password, **extra_fields)
+
+    def create_superuser(self, uid, password, **extra_fields):
+        extra_fields.setdefault('is_staff', True)
+        extra_fields.setdefault('is_superuser', True)
+
+        if extra_fields.get('is_staff') is not True:
+            raise ValueError("Superuser must have is_staff=True.")
+        if extra_fields.get('is_superuser') is not True:
+            raise ValueError('Superuser must have is_superuse=True.')
+        return self._create_user(uid, password, **extra_fields)
+
+
+class User(AbstractBaseUser, PermissionsMixin):
+    # Social needs the below setting. Username is not really set to UID.
+    USERNAME_FIELD = 'uid'
+    # Github user attributes.
+    uid = models.CharField(max_length=30, unique=True, blank=True, null=True)
+    login = models.CharField(max_length=50, blank=True, null=True)
```

Comment:
what is this?

Reply:
`Login` is Github's username, guess we don't really need it now as the pipeline is working. `UID` is from Github as well. We use it to verify that if a user is new or not.